src/main/java/com/minecolonies/coremod/entity/citizenhandlers/CitizenHappinessHandler.java

```diff
@@ -0,0 +1,596 @@
+     */
+    private double baseHappiness;
+
+    /**
+     * holds the modifier for food.
+     */
+    private double foodModifier;
+
+    /**
+     * holds the modifier for damage.
+     */
+    private double damageModifier;
+
+    /**
+     * holds the modifier for house.
+     */
+    private double houseModifier;
+
+    /**
+     * holds the numbers of days without a house.
+     */
+    private int numberOfDaysWithoutHouse;
+
+    /**
+     * holds the modifier for job.
+     */
+    private double jobModifier;
+
+    /**
+     * holds the numbers of days without a job.
+     */
+    private int numberOfDaysWithoutJob;
+
+    /**
+     * holds the modifier for farms that the farmer can't farm.
+     */
+    private double farmerModifier;
+
+    /**
+     * holds an indicator of the farmer has no fields to farm
+     */
+    private boolean hasNoFields;
+
+    /**
+     * holds an array of all the fields the farmer has assigned to them.
+     */
+    private final Map<BlockPos, FieldDataModifier> fieldModifier = new HashMap<BlockPos, FieldDataModifier>();
+
+    /**
+     * holds an indicator citizen needing a tool
+     */
+    private final Map<IToolType, Integer> needsTool = new HashMap<IToolType, Integer>();
+
+    /**
+     * holds the modifier for citizen not having a tool.
+     */
+    private double noToolModifier;
+
+    /**
+     * Constructor for the experience handler.
+     *
+     * @param citizen
+     *            the citizen owning the handler.
+     */
+    public CitizenHappinessHandler(final CitizenData citizen)
+    {
+        this.citizen = citizen;
+        baseHappiness = BASE_HAPPINESS;
+        foodModifier = 0;
+        damageModifier = 0;
+        houseModifier = 0;
+        numberOfDaysWithoutHouse = 0;
+        jobModifier = 0;
+        numberOfDaysWithoutJob = 0;
+        farmerModifier = 0;
+        hasNoFields = false;
+        needsTool.clear();
+        chatSpamFilter = new ChatSpamFilter(citizen);
+
+    }
+
+    /**
+     * This function applies eating adjust to the base hapiness for
+     * the citizen.
+     *
+     * @param eatFood
+     *            true or false indicate citizen was unable to eat
+     */
+    public void setFoodModifier(final boolean eatFood)
+    {
+        if (!eatFood)
+        {
+            if (citizen.getSaturation() < CitizenConstants.LOW_SATURATION)
+            {
+                foodModifier = FOOD_MODIFIER_MAX;
+            }
+            else
+            {
+                foodModifier = FOOD_MODIFIER_MIN;
+            }
+        }
+        else
+        {
+            foodModifier = 0;
+        }
+    }
+
+    /**
+     * Called once a day to update the citizens daily happiness
+     * modifiers.
+     *
+     * @param hasHouse  boolean if the citizen is assigned to a house
+     * @param hasJob boolean to indicate if citizen has a job
+     */
+    public void processDailyHappiness(final boolean hasHouse, final boolean hasJob)
```

Comment:
### Errors:
- 'public' method processDailyHappiness() is not exposed via an interface